themes/classic/templates/catalog/_partials/product-cover-thumbnails.tpl

```diff
@@ -26,7 +26,7 @@
   {block name='product_cover'}
     <div class="product-cover">
       {if $product.cover}
-        <img class="js-qv-product-cover" src="{$product.cover.bySize.large_default.url}" alt="{$product.cover.legend}" title="{$product.cover.legend}" style="width:100%;" itemprop="image">
+        <img class="js-qv-product-cover" src="{$product.cover.bySize.large_default.url}" alt="{if !empty($product.cover.legend)}{$product.cover.legend}{else}{$product.name|truncate:30:'...'}{/if}" title="{if !empty($product.cover.legend)}{$product.cover.legend}{else}{$product.name|truncate:30:'...'}{/if}" style="width:100%;" itemprop="image">
```

Comment:
@dominictayloruk I have an another question : Why 30 and not 40 or 50 ?

Reply:
In response to @MathiasReker & @Progi1984 why truncate 30?
Well i just put a limit so there were no unwieldy descriptive texts here. After a little research;

https://www.hobo-web.co.uk/how-many-words-in-alt-text-for-google-yahoo-bing/

> SO….
>Google seemed to count the first 16 words in the ALT tag text – of the image on this current page – in this instance –  and interestingly in the snippet, Google uses it does seem to completely cut off the rest of the ALT and from the 17th word
Having 16 words to work with might prove very useful if you are using ALT tags to describe more complex images. There is potentially plenty of available space to describe images properly for accessibility purposes AND SEO impact.

https://moz.com/learn/seo/alt-text

> Keep it (relatively) short. The most popular screen readers cut off alt text at around 125 characters, so it's advisable to keep it to that character count or less.

This leads to the conclusion that it should definitely be increased to 100 or 125?
What are your thoughts?

Reply:
I would say 125.

Reply:
Agreed.
From both a usability point of view and for SEO purposes.